### PR TITLE
reportefacturaelectronica

### PR DESCRIPTION
**Pull request by cadsx**

### CODE REVIEW

File: app_control/views.py [hunk 1]
```diff
@@ -864,3 +864,61 @@ def post(self, request):
...
+
+class FacturaElectronicaExporter(APIView):
+    http_method_names = ('post',)
+    permission_classes = (IsAuthenticatedCustom,)
+
+    def post(self, request):
+        response = HttpResponse(content_type='application/vnd.openxmlformats-officedocument.spreadsheetml.sheet')
+        response['Content-Disposition'] = 'attachment; filename="DocumentosVentasEncabezadosMovimientoInventarioWO.xlsx"'
+
+        start_date = request.data.get("start_date", None)
```
Comment (ndf1511) [inline app_control/views.py:876]: timestamp instead of dates

Comment (ndf1511) [reply app_control/views.py:876]: Move dates to start of function. line 873

File: app_control/views.py [hunk 2]
```diff
@@ -864,3 +864,61 @@ def post(self, request):
...
         wb.save(response)
         return response
+
+class FacturaElectronicaExporter(APIView):
+    http_method_names = ('post',)
+    permission_classes = (IsAuthenticatedCustom,)
+
+    def post(self, request):
+        response = HttpResponse(content_type='application/vnd.openxmlformats-officedocument.spreadsheetml.sheet')
+        response['Content-Disposition'] = 'attachment; filename="DocumentosVentasEncabezadosMovimientoInventarioWO.xlsx"'
```
Comment (ndf1511) [inline app_control/views.py:874]: Put dates in filename

File: app_control/views.py [hunk 3]
```diff
@@ -864,3 +864,61 @@ def post(self, request):
...
+            return Response({"error": "You need to provide start_date and end_date"})
+
+        wb = Workbook()
+        ws = wb.active
+        ws.title = "QueryDef_Exportar"
+
+        world_office_report_data = (
+            Invoice.objects.select_related("invoice_number", "PaymentMethods", "payment_terminal", "InvoiceItems", "created_by")
+            .all()
+            .filter(created_at__date__gte=start_date, created_at__date__lte=end_date)
```
Comment (ndf1511) [inline app_control/views.py:889]: not date,  timestamp

File: app_control/views.py [hunk 4]
```diff
@@ -864,3 +864,61 @@ def post(self, request):
...
+            .filter(is_override=False)
```
Comment (ndf1511) [inline app_control/views.py:890]: .filter(invoice_items__is_gift=False)

File: inventory_api/utils.py [hunk 5]
```diff
@@ -492,3 +492,37 @@ def create_invoices_report(ws, report_data):
     # apply english accounting format to all columns from 6 to 11
     for row in range(2, len(report_data) + 2):
         ws[f"{get_column_letter(6)}{row}"].number_format = '_($* #,##0.00_);_($* (#,##0.00);_($* "-"??_);_(@_)'
+
+def world_office_report(ws, report_data):
+    row_titles = ["Empresa","Tipo Documento","prefijo","DocumentoNúmero","Fecha","Tercero Interno","Tercero Externo","Nota",
```
Comment (ndf1511) [inline inventory_api/utils.py:497]: column_titles

File: inventory_api/utils.py [hunk 6]
```diff
@@ -492,3 +492,37 @@ def create_invoices_report(ws, report_data):
...
+                          font=headers_font, alignment=alignment, fill=headers_fill, border=border_style)
+    for column in range(1, 50):
+        ws.column_dimensions[get_column_letter(column)].width = 20
+
+    for row_idx, row_data in enumerate(report_data, start=2):
+        for col_idx, cell_value in enumerate(row_data, start=1):
+            ws.cell(row=row_idx, column=col_idx, value=cell_value)
+
+    # apply english accounting format to all columns from 6 to 11
+    for cell in ws[get_column_letter(5)]:
```
Comment (ndf1511) [inline inventory_api/utils.py:515]: for cell in  [ws[], ws[]]

**Review by ndf1511 — CHANGES_REQUESTED** [review]

File: app_control/urls.py [hunk 7]
```diff
@@ -3,7 +3,7 @@
     DianResolutionView, InventoryView, SummaryView, InvoiceView, PurchaseView,
     InventoryGroupView, SalePerformance, InventoryCSVLoaderView, UpdateInvoiceView,
     PaymentTerminalView, ReportExporter, ProviderView, SalesByUsersView, InventoriesReportExporter, ItemsReportExporter,
-    InvoicesReportExporter, CustomerView, InvoicePainterView
+    InvoicesReportExporter, CustomerView, InvoicePainterView, FacturaElectronicaExporter
```
Comment (oskarkos) [inline app_control/urls.py:6]: Name in english

**Review by ndf1511 — APPROVED** [review]

**Pull request merged**